cmd/syncthing/main.go

```diff
@@ -216,7 +217,11 @@ func main() {
 	// The "cli" subcommand uses a different command line parser, and e.g. help
 	// gets mangled when integrating it as a subcommand -> detect it here at the
 	// beginning.
-	if len(os.Args) > 1 && os.Args[1] == "cli" {
+	doCompgen := (strings.Contains(os.Getenv("COMP_LINE"), " cli") && os.Getenv("SHELL") == "/bin/bash")
```

Comment:
Some explanation on what the intent of these checks would be nice. Why do we care if its bash or sh?